symphony-bdk-core/src/main/java/com/symphony/bdk/core/auth/AuthenticatorFactory.java

```diff
@@ -94,28 +119,47 @@ OboAuthenticator getOboAuthenticator() throws AuthInitializationException {
   public @Nonnull
   ExtensionAppAuthenticator getExtensionAppAuthenticator() throws AuthInitializationException {
     if (this.config.getApp().isCertificateAuthenticationConfigured()) {
+      if (!this.config.getApp().isCertificateConfigurationValid()) {
+        throw new AuthInitializationException(
+            "Only one of certificate path or content should be configured for app authentication.");
+      }
       return new ExtensionAppAuthenticatorCertImpl(
           this.config.getApp().getAppId(),
           this.apiClientFactory.getExtAppSessionAuthClient());
     }
-    return new ExtensionAppAuthenticatorRsaImpl(
-        this.config.getApp().getAppId(),
-        this.loadPrivateKeyFromPath(this.config.getApp().getPrivateKeyPath()),
-        this.apiClientFactory.getLoginClient(),
-        this.apiClientFactory.getPodClient()
-    );
+    if (this.config.getApp().isRsaAuthenticationConfigured()) {
+      if (!this.config.getApp().isRsaConfigurationValid()) {
+        throw new AuthInitializationException(
+            "Only one of private key path or content should be configured for app authentication.");
+      }
+      return new ExtensionAppAuthenticatorRsaImpl(
+          this.config.getApp().getAppId(),
+          this.loadPrivateKeyFromAuthenticationConfig(this.config.getApp()),
+          this.apiClientFactory.getLoginClient(),
+          this.apiClientFactory.getPodClient()
+      );
+    }
+    throw new AuthInitializationException("Neither RSA private key nor certificate is configured.");
   }
 
-  private PrivateKey loadPrivateKeyFromPath(String privateKeyPath) throws AuthInitializationException {
-    log.debug("Loading RSA privateKey from path : {}", privateKeyPath);
+  private PrivateKey loadPrivateKeyFromAuthenticationConfig(BdkAuthenticationConfig config)
+      throws AuthInitializationException {
     try {
-      return this.jwtHelper.parseRsaPrivateKey(loadPrivateKey(privateKeyPath));
+      String privateKey;
+      if (isNotEmpty(config.getPrivateKeyContent())) {
+        privateKey = new String(config.getPrivateKeyContent());
```

Comment:
Here the default platform charset will be used instead of a fixed one, maybe it would be more predictive to always use UTF-8 and document that the bytes should be set to UTF-8